workflow/scores/curves.py

```diff
@@ -0,0 +1,208 @@
+            [(x, y, r) for (x, y, r, p) in y_pred_patch if p > conf_threshold]
+            for y_pred_patch in y_pred]
+        ps.append(precision(y_true, y_pred_temp, iou_threshold=iou_threshold))
+        rs.append(recall(y_true, y_pred_temp, iou_threshold=iou_threshold))
+
+    return np.array(ps), np.array(rs)
+
+
+def mask_detection_curve(y_true, y_pred, conf_thresholds):
+    """
+    Calculate mask detection score for different confidence thresholds.
+
+    Parameters
+    ----------
+    y_true : list of list of tuples
+        Tuples are of form (x, y, radius).
+    y_pred : list of list of tuples
+        Tuples are of form (x, y, radius, confidence).
+    conf_thresholds : array-like
+        The confidence threshold for which to calculate the
+        precision and recall.
+
+    Returns
+    -------
+    ms : array with score for each confidence threshold
+
+    """
+    from .mask import mask_detection
+
+    ms = []
+
+    for conf_threshold in conf_thresholds:
+        y_pred_temp = [
+            [(x, y, r) for (x, y, r, p) in y_pred_patch if p > conf_threshold]
+            for y_pred_patch in y_pred]
+        ms.append(mask_detection(y_true, y_pred_temp))
+
+    return np.array(ms)
+
+
+def ospa_curve(y_true, y_pred, conf_thresholds):
+    """
+    Calculate OSPA score for different confidence thresholds.
+
+    Parameters
+    ----------
+    y_true : list of list of tuples
+        Tuples are of form (x, y, radius).
+    y_pred : list of list of tuples
+        Tuples are of form (x, y, radius, confidence).
+    conf_thresholds : array-like
+        The confidence threshold for which to calculate the
+        precision and recall.
+
+    Returns
+    -------
+    os : array with OSPA score for each confidence threshold
+
+    """
+    from .ospa import ospa
+
+    os = []
+
+    for conf_threshold in conf_thresholds:
+        y_pred_temp = [
+            [(x, y, r) for (x, y, r, p) in y_pred_patch if p > conf_threshold]
+            for y_pred_patch in y_pred]
+        os.append(ospa(y_true, y_pred_temp))
+
+    return np.array(os)
+
+
+def average_precision_interpolated(ps, rs):
+    """
+    The Average Precision (AP) score.
+
+    Calculation based on the 11-point interpolation of the precision-recall
+    curve (method described for Pascal VOC challenge,
+    http://homepages.inf.ed.ac.uk/ckiw/postscript/ijcv_voc09.pdf).
+
+    TODO: they changed this in later: http://homepages.inf.ed.ac.uk/ckiw/postscript/ijcv_voc09.pdf
+
+    https://stackoverflow.com/questions/36274638/map-metric-in-object-detection-and-computer-vision
+
+    Parameters
+    ----------
+    ps, rs : arrays of same length with corresponding precision / recall scores
+
+    Returns
+    -------
+    ap : int [0 - 1]
+        Average precision score
+
+    """
+    ps = np.asarray(ps)
+    rs = np.asarray(rs)
+
+    p_at_r = []
+
+    for r in np.arange(0, 1.1, 0.1):
+        p = np.array(ps)[np.array(rs) >= r]
+        if p.size:
```

Comment:
If I want to be peeky (there's nothing more to say, so..) I would rather use `np.linspace(0, 1, 11)` 😃